ydb/mvp/oidc_proxy/oidc_protected_page_nebius.h

```diff
@@ -0,0 +1,155 @@
+        if (!event->Get()->Response) {
+            LOG_DEBUG_S(ctx, EService::MVP, "Getting access token: Bad Request");
+            NHttp::THeadersBuilder ResponseHeaders;
+            ResponseHeaders.Set("Content-Type", "text/plain");
+            NHttp::THttpOutgoingResponsePtr httpResponse = Request->CreateResponse("400", "Bad Request", ResponseHeaders, event->Get()->Error);
+            ctx.Send(Sender, new NHttp::TEvHttpProxy::TEvHttpOutgoingResponse(httpResponse));
+            Die(ctx);
+        } else {
+            NHttp::THttpIncomingResponsePtr response = event->Get()->Response;
+            LOG_DEBUG_S(ctx, EService::MVP, TStringBuilder() << "Getting access token: " << response->Status);
+            if (response->Status == "200") {
+                TString iamToken;
+                static NJson::TJsonReaderConfig JsonConfig;
+                NJson::TJsonValue requestData;
+                bool success = NJson::ReadJsonTree(response->Body, &JsonConfig, &requestData);
+                if (success) {
+                    iamToken = requestData["access_token"].GetStringSafe({});
+                }
+
+                const TString authHeader = IAM_TOKEN_SCHEME + iamToken;
+                ForwardUserRequest(authHeader, ctx);
+            } else {
+                if (response->Status == "400" || response->Status == "401") {
+                    StartOidcProcess(ctx);
+                    RequestAuthorizationCode(ctx);
+                } else {
+                    NHttp::THttpOutgoingResponsePtr httpResponse;
+                    NHttp::THeadersBuilder ResponseHeaders;
+                    ResponseHeaders.Parse(response->Headers);
+                    httpResponse = Request->CreateResponse(response->Status, response->Message, ResponseHeaders, response->Body);
+                    ctx.Send(Sender, new NHttp::TEvHttpProxy::TEvHttpOutgoingResponse(httpResponse));
+                    Die(ctx);
+                }
+            }
+        }
+    }
+
+    STFUNC(StateWork) {
+        switch (ev->GetTypeRewrite()) {
+            HFunc(NHttp::TEvHttpProxy::TEvHttpIncomingResponse, HandleProxy);
+        }
+    }
+
+    STFUNC(StateExchange) {
+        switch (ev->GetTypeRewrite()) {
+            HFunc(NHttp::TEvHttpProxy::TEvHttpIncomingResponse, HandleExchange);
+        }
+    }
+
+private:
+    void ExchangeSessionToken(const TString sessionToken, const NActors::TActorContext& ctx) {
+        LOG_DEBUG_S(ctx, EService::MVP, "Exchange session token");
+        NHttp::THttpOutgoingRequestPtr httpRequest = NHttp::THttpOutgoingRequest::CreateRequestPost(Settings.GetExchangeEndpointURL());
+        httpRequest->Set<&NHttp::THttpRequest::ContentType>("application/x-www-form-urlencoded");
+
+        NMVP::TMvpTokenator* tokenator = MVPAppData()->Tokenator;
+        TString token = "";
+        if (tokenator) {
+            token = tokenator->GetToken(Settings.SessionServiceTokenName);
```

Comment:
What if this token is empty? It it error?

Reply:
I would say it's unexpected behaviour. We will see `Getting access token: 401` in the logs than

Reply:
Then it's worth to validate?